docs/src/extend/stats.md

```diff
@@ -0,0 +1,116 @@
+---
+title: Stats Data
+eleventyNavigation:
+    key: Stats Data
+    parent: extend eslint
+    title: Stats Data
+---
+
+While an analysis of the overall rule performance for an ESLint run can be carried out by setting the [TIMING](./custom-rules#profile-rule-performance) environment variable, it can sometimes be useful to acquire more *granular* timing data (lint time per file per rule) or collect other measures of interest. In particular, when developing new [custom plugins](./plugins) and evalulating/benchmarking new languages or rule sets.
+
+For such use cases, the **stats** option has been provided. It adds a series of detailed performance statistics (see [Stats type](#-stats-type)) such as the *parse*-, *fix*- and *lint*-times (time per rule) of a given file or the number of fix passes on top of your [LintResult](../integrate/nodejs-api#-lintresult-type).
+
+As such, it is not available via stdout but made easily ingestible via a formatter using the CLI or via the Node.js API to cater to your specific needs.
+
+## ◆ Stats type
+
+The `Stats` value is the timing information of each lint run. The `stats` property of the [LintResult] type contains it. It has the following properties:
+
+* `fixPasses` (`number`)<br>
+  The number of times ESLint has applied at least one fix after linting.
+* `times` (`TimePass[]`)<br>
+  The times spent on (parsing, fixing, linting) a file, where the linting refers to the timing information for each rule.
+    * `TimePass` (`{ parse: ParseTime, rules: Record<string, RuleTime> fix: FixTime }`)<br>
+    An array containing the times spent on (parsing, fixing, linting)
+    * `ParseTime` (`{ total: number }`)<br>
+      The total time that is spent when parsing a file.
+    * `RuleTime` (`{ total: number }`)<br>
+      The total time that is spent on a rule.
+    * `FixTime` (`{ total: number }`)<br>
+      The total time that is spent on applying fixes to the code.
+
+### CLI usage
+
+Let's consider the following example:
+
+```js [file-to-fix.js]
+/*eslint no-regex-spaces: "error", wrap-regex: "error"*/
+
+function a() {
+    return / foo/.test("bar");
+}
+```
+
+Run ESLint with the *stats* option (`--stats`) and outputting to JSON via a [built-in formatter](../use/formatters/) (`-f json`):
+
+```sh
+eslint file-to-fix.js --stats -f json
```

Comment:
Since the file is named "file-to-fix", it might be more representative to run eslint with `--fix` and get an array with multiple elements in `passes`?